tuning/utils/preprocessing_utils.py

```diff
@@ -125,10 +173,7 @@ def get_data_collator(
             )
         # Note that this automatically pads labels with -100
         # TODO check if this is sufficient for preprocessed
-        if (
-            "attention_mask" in formatted_train_dataset.column_names
-            and "labels" in formatted_train_dataset.column_names
-        ):
+        if is_train_data_pretokenized:
```

Comment:
packing is anyway not supported for pretokenized as well, so I just moved it here @kmehant like it was before. The check was doing same thing as your function is_pretokenized. 

padding=True is same as padding = longest . see https://huggingface.co/transformers/v4.9.2/main_classes/data_collator.html

Reply:
Thanks